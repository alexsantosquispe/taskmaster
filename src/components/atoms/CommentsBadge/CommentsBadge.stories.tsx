import type { Meta, StoryObj } from '@storybook/react';

import { CommentsBadge } from './CommentsBadge';

const meta = {
  title: 'Atoms/CommentsBadge',
  component: CommentsBadge,
  parameters: {
    layout: 'centered'
  }
} satisfies Meta<typeof CommentsBadge>;

export default meta;

type Story = StoryObj<typeof CommentsBadge>;

export const Default: Story = () => {
  return <CommentsBadge numberOfComments={26} />;
};

Default.args = {};
