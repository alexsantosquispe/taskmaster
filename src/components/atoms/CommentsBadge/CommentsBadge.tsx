import { twMerge } from 'tailwind-merge';
import { CommentCircleIcon } from '../../../icons';

interface CommentsBadgeProps {
  numberOfComments: number;
  className?: string;
}

export const CommentsBadge = ({
  numberOfComments,
  className
}: CommentsBadgeProps) => {
  return (
    <div
      className={twMerge(
        'flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80',
        className
      )}
    >
      <CommentCircleIcon className="text-neutral-400" />
      <span>{numberOfComments}</span>
    </div>
  );
};
