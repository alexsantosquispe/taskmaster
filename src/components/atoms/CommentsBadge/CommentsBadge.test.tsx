import { CommentsBadge } from './CommentsBadge';
import { render } from '@testing-library/react';

describe('CommentsBadge', () => {
  it('should render the component correctly', () => {
    const component = render(<CommentsBadge numberOfComments={10} />);

    expect(component).toMatchSnapshot();
  });
});
