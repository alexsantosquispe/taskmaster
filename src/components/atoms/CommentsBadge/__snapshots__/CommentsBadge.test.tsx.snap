// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CommentsBadge should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
      >
        <svg
          class="size-4 text-neutral-400"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
          />
          <path
            d="M8 12h.01"
          />
          <path
            d="M12 12h.01"
          />
          <path
            d="M16 12h.01"
          />
        </svg>
        <span>
          10
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex items-center justify-center gap-1 rounded-md border border-neutral-200 px-1 py-0.5 font-semibold text-neutral-600 shadow-sm dark:border-neutral-700 dark:text-white/80"
    >
      <svg
        class="size-4 text-neutral-400"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.9 20A9 9 0 1 0 4 16.1L2 22Z"
        />
        <path
          d="M8 12h.01"
        />
        <path
          d="M12 12h.01"
        />
        <path
          d="M16 12h.01"
        />
      </svg>
      <span>
        10
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
